src/role-access/containers/exclusion/choose-person-role/choose-person-role.component.ts

```diff
@@ -1,46 +1,91 @@
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { select, Store } from '@ngrx/store';
+import { Subscription } from 'rxjs';
 import { Observable } from 'rxjs/Observable';
-import { RoleAllocationCaptionText, RoleAllocationTitleText } from 'src/role-access/models/enums';
-
-import { ExclusionNavigationEvent, ExclusionState, RadioOption, Role } from '../../../models';
+import { ERROR_MESSAGE, PERSON_ROLE } from '../../../constants';
+import { ExclusionNavigationEvent, ExclusionState, PersonRole, Role } from '../../../models';
+import { RoleAllocationCaptionText, RoleAllocationTitleText } from '../../../models/enums';
 import { ExclusionNavigation } from '../../../models/exclusion-navigation.interface';
-import { RoleExclusionsService } from '../../../services/role-exclusions.service';
+import { OptionsModel } from '../../../models/options-model';
+import { RoleExclusionsService } from '../../../services';
 import * as fromFeature from '../../../store';
 
 @Component({
   selector: 'exui-choose-person-role',
   templateUrl: './choose-person-role.component.html',
   styleUrls: ['./choose-person-role.component.scss']
 })
-export class ChoosePersonRoleComponent implements OnInit {
-
+export class ChoosePersonRoleComponent implements OnInit, OnDestroy {
+  public ERROR_MESSAGE = ERROR_MESSAGE;
   @Input() public navEvent: ExclusionNavigation;
-  public roleOptions: RadioOption[];
+
   public roles$: Observable<Role[]>;
   public title = RoleAllocationTitleText.ExclusionChoose;
   public caption = RoleAllocationCaptionText.Exclusion;
+  public optionsList: OptionsModel[];
+
+  public submitted: boolean = false;
+
+  public formGroup: FormGroup;
+  public radioOptionControl: FormControl;
+  public radioControlName: string = PERSON_ROLE;
 
-  constructor(private readonly store: Store<fromFeature.State>, private readonly roleExclusionsService: RoleExclusionsService) { }
+  public exclusionStateDataSub: Subscription;
+
+  public personRole: PersonRole;
+
+  constructor(private readonly store: Store<fromFeature.State>,
+              private readonly roleExclusionsService: RoleExclusionsService) {
+  }
 
   public ngOnInit(): void {
+    this.exclusionStateDataSub = this.store.pipe(select(fromFeature.getRoleAccessState)).subscribe(
+      exclusionStateData => {
+        this.personRole = exclusionStateData.personRole;
+      }
+    );
+
+    this.radioOptionControl = new FormControl(this.personRole ? this.personRole : '', [Validators.required]);
+    this.formGroup = new FormGroup({[this.radioControlName]: this.radioOptionControl});
+
     this.roles$ = this.roleExclusionsService.getRolesCategory();
     this.roles$.subscribe((roles) => {
-      this.roleOptions = [];
-      roles.forEach(role => {
-        const currentOption: RadioOption = { radioId: role.roleId, radioName: role.roleName };
-        this.roleOptions.push(currentOption);
+      this.optionsList = roles.map(role => {
```

Comment:
Could be inline..

Reply:
this has been indented correctly, it aligns with the original roles.forEach...